portalnet/src/utp_controller.rs

```diff
@@ -15,33 +25,253 @@ use crate::discovery::UtpEnr;
+        self.metrics
+            .report_utp_active_inc(UtpDirectionLabel::Outbound);
+        let stream = match self.connect_with_cid(cid.clone(), *UTP_CONN_CFG).await {
+            Ok(stream) => stream,
+            Err(err) => {
+                self.metrics.report_utp_outcome(
+                    UtpDirectionLabel::Outbound,
+                    UtpOutcomeLabel::FailedConnection,
+                );
+                debug!(
+                    %err,
+                    cid.send,
+                    cid.recv,
+                    peer = ?cid.peer.client(),
+                    "Unable to establish uTP conn based on Accept",
+                );
+                return false;
+            }
+        };
+
+        // send_utp_content handles metrics reporting of successful & failed txs
+        match Self::send_utp_content(stream, &data, self.metrics.clone()).await {
+            Ok(_) => true,
+            Err(err) => {
+                debug!(
+                    %err,
+                    %cid.send,
+                    %cid.recv,
+                    peer = ?cid.peer.client(),
+                    "Error sending content over uTP, in response to ACCEPT"
+                );
+                false
+            }
+        }
+    }
+
+    /// Accept an outbound stream given the connection id, and transfer the data to the peer.
+    pub async fn accept_outbound_stream(
+        &self,
+        cid: ConnectionId<UtpEnr>,
+        content_id: [u8; 32],
+        data: Vec<u8>,
+    ) {
+        self.metrics
+            .report_utp_active_inc(UtpDirectionLabel::Outbound);
+        let stream = match self.accept_with_cid(cid.clone(), *UTP_CONN_CFG).await {
+            Ok(stream) => stream,
+            Err(err) => {
+                self.metrics.report_utp_outcome(
+                    UtpDirectionLabel::Outbound,
+                    UtpOutcomeLabel::FailedConnection,
+                );
+                debug!(
+                    %err,
+                    %cid.send,
+                    %cid.recv,
+                    peer = ?cid.peer.client(),
+                    "unable to accept uTP stream for CID"
+                );
+                return;
+            }
+        };
+        // send_utp_content handles metrics reporting of successful & failed txs
+        if let Err(err) = Self::send_utp_content(stream, &data, self.metrics.clone()).await {
+            debug!(
+                %err,
+                %cid.send,
+                %cid.recv,
+                peer = ?cid.peer.client(),
+                content_id = %hex_encode(content_id),
+                "Error sending content over uTP, in response to FindContent"
+            );
+        };
+    }
+
+    /// Accept an inbound stream given the connection id, and return the data received from the
+    /// peer.
+    pub async fn accept_inbound_stream(
+        &self,
+        cid: ConnectionId<UtpEnr>,
+        content_keys_string: Vec<String>,
```

Comment:
same I think we should just remove this log

Reply:
you want to remove both debug logs that use this var?

Reply:
i'm leaning towards leaving (all the debug logs) in, but just removing the `content_keys_string` and `content_id` vars from those logs

Reply:
nvm, we can actually just remove them since we emit a `debug!` log inside `OverlayService` if an error is returned

Reply:
If I'm not mistaken, the caller already logs if there is failure.
I think this function can just return `Err` with message as a reason, and caller can log all details, including the reason.